### PR TITLE
LNK-1360: Reporting metrics rest controller

### PR DESCRIPTION
**Pull request by amphillipsLGC**
Added a new metrics rest controller along with methods to calculate the various metrics needed for reporting.

### CODE REVIEW

File: api/src/main/java/com/lantanagroup/link/api/controller/MetricController.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/metric")
+public class MetricController extends BaseController {
+  private static final Logger logger = LoggerFactory.getLogger(MetricController.class);
+  private static final String QUERY_CATEGORY = "query";
```
Comment (seanmcilvenna) [inline api/src/main/java/com/lantanagroup/link/api/controller/MetricController.java:23]: Move these to the Constants class please. These may be used by other code I suspect.

File: api/src/main/java/com/lantanagroup/link/api/controller/MetricController.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    //calculate metrics based on period, we can assume that all categories have already been
+    //filtered down based on tenant and/or report if those parameters were provided
+    MetricsReportResponse report = this.CalculatePeriodMetrics(period, endDate, metrics);
+
+    return report;
+  }
+
+  //TODO: This functionality probably should be broken out into a service
+
+  private MetricsReportResponse CalculatePeriodMetrics(String period, LocalDate end, List<Metrics> metrics) {
```
Comment (seanmcilvenna) [inline api/src/main/java/com/lantanagroup/link/api/controller/MetricController.java:114]: Java is camel case method naming..
`calculatePeriodMetrics()`

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]
Only some syntax complaints. Otherwise looks super solid. Going to checkout the branch later and try it out, too. Happy to merge as is though, too.

File: api/src/main/java/com/lantanagroup/link/api/controller/MetricController.java (hunk 2 above)
Comment (amphillipsLGC) [reply api/src/main/java/com/lantanagroup/link/api/controller/MetricController.java]: Eww but fixed haha.

**Review by seanmcilvenna — APPROVED** [review]

**Pull request merged**